torch/nn/modules/module.py

```diff
@@ -2103,7 +2103,8 @@ def load_state_dict(self, state_dict: Mapping[str, Any],
 
         .. warning::
             If :attr:`assign` is ``True`` the optimizer must be created after
-            the call to :attr:`load_state_dict`.
+            the call to :attr:`load_state_dict` unless
+            :func:`~torch.__future__.get_swap_module_params_on_conversion` is ``True``.
```

Comment:
Would appreciate feedback on how to make the behavior of `assign` when swap is set more clear in the docstring for the argument :)

Reply:
The way I think about this is: one preserves the properties of the Tensor currently in the model while the other preserves the properties of the Tensor currently in the state dict.

Reply:
Agreed! And we do document it as such!

 > assign (bool, optional): whether to assign items in the state
                dictionary to their corresponding keys in the module instead
                of copying them inplace into the module's current parameters and buffers.
                When ``False``, the properties of the tensors in the current
                module are preserved while when ``True``, the properties of the
                Tensors in the state dict are preserved.
                Default: ``False`

I do feel that if I was not familiar with the internals, it would be a little strange to reason about what swapping means in conjunction with `assign` though. In particular, do you think there is a better way to word

> assign items in the state dictionary to their corresponding keys in the module

for the swap case, or do you think this seems fine as it is

Also one small edge case is: do we consider `requires_grad` a property of the tensor? :P

Reply:
We should definitely callout the requires_grad special behavior here.

I do think that the doc can be simplified here. We go a bit too much in the details of how its implemented maybe? If we do need a 100% accurate description, we most likely need a longer form comment anyways that mentions how each is implemented.
So maybe I would just remove the first sentence there and only keep "When False, the properties of the tensors in the current module are preserved while when True, the properties of the Tensors in the state dict are preserved.".